[Keyboard] Add CZC.Gaming Dwarf

Vesek opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description
Added a new keyboard, it is only sold regionally but there might be some people searching for this so might as well push it here.
<!--- Describe your changes in detail here. -->


## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] Functioning keyboard
- [x] Full original functionality ~~(It's just macros, might not do)~~ Turns out this keyboard didn't have dedicated key combos for macros (never used it), so less work for me :) Edit: And Bluetooth doesn't work
- [x] RGB
- [x] Via

Vesek committed:
[Keyboard] Add CZC.Gaming Dwarf

Vesek committed:
Add RGB matrix support

Vesek committed:
Misc fixes and new VID/PID

Vesek committed:
Add VIA support
A lot of help came from designedbygg/berserk config so thanks to it's maintainer

Vesek committed:
Enable all matrix effects

dexter93 commented on keyboards/czc/dwarf/readme.md:
```diff
@@ -0,0 +1,23 @@
+
+Make example for this keyboard (after setting up your build environment):
+
+    make czc/dwarf:default
```
this is old style. switching to the new way is advisable
```suggestion
    qmk compile -kb czc/dwarf -km default
```

Vesek replied:
Actually didn't know about this one, will modify.

dexter93 commented on keyboards/czc/dwarf/readme.md:
```diff
@@ -0,0 +1,23 @@
+
+Flashing example for this keyboard:
+
+    This keyboard must be flashed with an external tool (https://github.com/SonixQMK/sonix-flasher)
```
you might want to try the cli variant, SonixFlasherC :)

Vesek replied:
I am using it since the beginning because it's faster to use and also the GUI always SIGFAULTs.
I will modify the readme.md to show a link to both

dexter93 commented on keyboards/czc/dwarf/readme.md:
```diff
@@ -0,0 +1,23 @@
+
+## Bootloader
+
+* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (Escape) and plug in the keyboard
```
or, use the `QK_BOOT` keycode at any moment

Vesek committed:
Change readme.md

Vesek committed:
Fix RGB matrix coords

Vesek committed:
Change readme.md

Vesek committed:
Change readme.md

Vesek committed:
Change the center of the RGB matrix
Maybe it's just me but i think the alphanumeric keys should be more of a centerpiece.

Vesek committed:
Remove the new center
It magically fixed itself?! (probably user error)

Vesek committed:
Change some RGB Matrix flags
Caps was flaged as an alphanumeric key and semicolon wasn't

dexter93 commented on keyboards/czc/dwarf/keymaps/via/keymap.c:
```diff
@@ -0,0 +1,55 @@
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,             RGB_VAI,
+        _______, GU_TOGG, _______,                            _______,                            _______, _______, _______, _______,    RGB_SPD, RGB_VAD, RGB_SPI
+    ),
+    [2] = LAYOUT_tkl_ansi(
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,             _______,
+        _______, _______, _______,                            _______,                            _______, _______, _______, _______,    _______, _______, _______
+    ),
+    [3] = LAYOUT_tkl_ansi(
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,    _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,
+        _______,          _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,             _______,
+        _______, _______, _______,                            _______,                            _______, _______, _______, _______,    _______, _______, _______
+    )
```
extra blank layers should not be needed anymore on VIA builds. Build system should take care of it, leaving keymap files cleaner
```suggestion
    )
```

Vesek committed:
Update VIA keymap.c

Co-authored-by: dexter93 <d3xter93@gmail.com>

3top1a commented:
Hi, do you have a firmware dump in any case?

Vesek commented:
@3top1a I do not, in the process of trying to obtain it i accidentaly erased it. So the easiest way to fix my keyboard was to flash QMK on it.

3top1a commented:
> @3top1a I do not, in the process of trying to obtain it i accidentaly erased it. So the easiest way to fix my keyboard was to flash QMK on it.

thanks. Did you have to short the boot pin to get into the bootloader?
Clicking reboot to bootloader [HFD] did something with mine

EDIT: When I clicked the button the RGB went off but the gui did not recognize it in bootloader mode, shorting the pins worked normally.
